api/api.ts

```diff
@@ -50,11 +50,11 @@ app.post('/upload/:tenant', async (req: express.Request, res: express.Response)
     }
 
     const filename: string = getFilename(req)
-    const tenantConfig = await getTenantConfig(req.params.tenant)
+    const tenantConfig = await getTenantConfig(req.params.tenant, req.headers['driver'] as string)
 
     if (!tenantConfig) {
-        console.error(`Tenant not found at ${new Date()}`)
-        res.status(404).send('Tenant not found.')
+        console.error(`Tenant or driver not found at ${new Date()}`)
```

Comment:
Could we work with the exact cause of the error instead of the "or"? Instead of returning null, you could return an `Error Type` which could tell us the exact cause of the error.

Reply:
I still don't know how to treat the following case: tenant exists, but the driver that was passed in header doesn't belong to this tenant. 
Your suggestion is about returning an Error typeinstead of `null` in `getTenantConfig` ?

Reply:
Yes, return an Error type instead of null in getTenantConfig. But I don't know if its easy to do with mongo. If you find it hard we could solve the conversation with a issue an move on.

Reply:
> Yes, return an Error type instead of null in getTenantConfig. But I don't know if its easy to do with mongo. If you find it hard we could solve the conversation with a issue an move on.

I guess it's easy to implement. We can do this in this PR.  

Just break this conjunction `tenantDoc && tenantDoc.drivers` in two separated `if` and returning the properly error in each `if`.

I'll do it in this PR yet : )

Reply:
I couldn't return an error, but I treated in `api.ts` . Could you review this new approach and resolve conversation if it's all ok ?